File: examples/benchmark/reader.py
import time
import json
from karait import Queue
print("Starting python reader.")

messages_read = 0.0
start_time = time.time()

queue = Queue()
while True:
  messages = queue.read(routing_key='for_reader', messages_read=15)
  for message in messages:
    messages_read += 1.0
        
    message.messages_read = messages_read
    message.messages_read_per_second = messages_read / (time.time() - start_time)
    
    if (messages_read % 250) == 0.0:
      print("Message Read: \n%s" % json.dumps(message.to_dictionary()))

  queue.delete_messages(messages)

File: examples/benchmark/reader.js
var puts = require('sys').puts,
    Queue = require('karait').Queue;

puts("Starting javascript reader.")

messagesRead = 0.0
startTime = (new Date()).getTime() / 1000.0;

new Queue(function(err, queue) {
    if (err) {
        throw err;
    }
    
    (function readMessages() {
        queue.read({routingKey: 'for_reader'}, function(err, messages) {
            for (var i = 0, message; (message = messages[i]) != null; i++) {
                messagesRead += 1;
                message.messages_read = messagesRead;
                message.messages_read_per_second = messagesRead / ( ( (new Date()).getTime() / 1000.0 ) - startTime );
                
                if (messagesRead % 250 == 0) {
                    puts(JSON.stringify(message.toObject()));
                }
            }
            queue.deleteMessages(messages, function() {
                readMessages();
            });
        });
    })();
});
